### PR TITLE
[MRG] Create a basic demo site

### LINKED ISSUE #242: [FEATURE] Demo site
There should be a live, public example of ModelDB for people to play with. Visitors should get to browse both a Jupyter notebook with example models to run and ModelDB's Node.js frontend to see reports about those models.

Architecture:

- Single ModelDB server
    - Mongo server
    - ModelDB Java backend
    - ModelDB Node.js frontend
- Multiple ephemeral Jupyter notebooks with ModelDB python library
    - [configurable-http-proxy](https://github.com/jupyterhub/configurable-http-proxy) to vend out notebooks
    - [tmpnb](https://github.com/jupyter/tmpnb) to spawn new notebooks and cull idle and old ones

This will also require the creation of one or more .ipynb files that can be run by users to create reports in ModelDB.

### PR DESCRIPTION
**Pull request by justinanderson**
Resolves #242

The site will live on modeldb.csail.mit.edu. Once this PR is closed, I'll update gh-pages with links to the demo server and notebook server and upload a build to Docker Hub of the Jupyter notebook used by the demo.

- [x] ModelDB site lives at :3000.
- [x] Jupyter notebooks are vended out from :8000.
- [x] Spawns 10 concurrent Jupyter notebooks.
- [x] Notebooks include ModelDB Python client library and its dependencies.
- [x] Notebooks idle for an hour are killed. Idleness is checked every 10 minutes.
- [x] ModelDB restarts every 24 hours and resets to a known state.
- [x] Site has preloaded demo data so it's not blank when the server restarts.
- [x] Site has multiple .ipynb files showing example usage of ModelDB's scikit-learn integration.

You can play around with an in-progress version of this now at http://modeldb.csail.mit.edu:3000 and http://modeldb.csail.mit.edu:8000. Only BasicSyncAll has been ported to a .ipynb file so far, but it's something to play with.

### CODE REVIEW

**Comment by justinanderson** [comment]
@mpvartak Can you take a look and tell me what you think? I think all that's left is to get some demo data to prefill the demo site with and to port more of the Python samples over to .ipynb files.

**Comment by mpvartak** [comment]
Tested out the site and the iPynb. Looks great! I'll send the current best DB dump. we can iterate on it more

**Comment by justinanderson** [comment]
@mpvartak I added some of the Python samples as ipynb files to the demo. The ones I didn't add seemed to either do nothing by themselves or be too complex for a demo site. I can add them if you want.

Otherwise I think this is ready to merge.

**Comment by mpvartak** [comment]
This looks good. Ok to merge. I'll open an Issue for myself to put in examples with TF or torch.

File: demo/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# ModelDB Demo Site
```
Comment (mpvartak) [inline demo/README.md:1]: Please add a comment here that these instructions are for creating the demo setup as on modeldb.csail.mit.edu. To set up a ModelDB installation on their local machine, point to the root README with those instructions.

**Review by mpvartak — APPROVED** [review]
Looks good. One minor doc comment

**Comment by mpvartak** [comment]
Also, @justinanderson Can you add a demo section to the root README that points to the demo server(s).

**Pull request merged**